src/main/java/net/neoforged/neoforge/client/event/CalculateDetachedCameraDistanceEvent.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import net.minecraft.client.Camera;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.BlockGetter;
+import net.neoforged.bus.api.Event;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.fml.LogicalSide;
+import net.neoforged.neoforge.common.NeoForge;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Fired in {@linkplain Camera#setup(BlockGetter, Entity, boolean, boolean, float) Camera#setup(BlockGetter, Entity, boolean, boolean, float)} when camera is detached before calculating the distance of the camera from the {@linkplain Camera#getEntity() camera entity} based on a hard-coded maximum and a raycast.
+ *
```

Comment:
I find this explanation a bit unclear. Would be nice to use simpler words, similar to "How far behind the player is the camera in 3rd person view?". Also please break that line.

Reply:
```
 * Fired for hooking the maximum distance from the player to the camera in third-person view.
 * The ray-cast that reduces this distance based on the blocks around the player is invoked after this event is fired.
```

Is this better?